src/services/api.js

```diff
@@ -1,8 +1,18 @@
 import axios from 'axios'
+import {
+  setLocalStorage,
+  getLocalStorage,
+  USER_DATA_KEY,
+  TOKEN_KEY,
+} from './localStorage'
 
 const axiosInstance = () => {
+  const token = getLocalStorage(TOKEN_KEY)
+  const headers = token ? { Authorization: `Bearer ${token}` } : undefined
   return axios.create({
     baseURL: `${process.env.GATSBY_STRAPI_API_URL}`,
+    timeout: 10000,
+    headers,
```

Comment:
actually I am not sure if headers = undefined works to remove the headers .. maybe split the assignments like 

```js
let params = {
      baseURL: `${process.env.GATSBY_STRAPI_API_URL}`,
     timeout: 10000, 
}
if (headers) {
  params.headers = headers
}
return axios.create(params)
```